R/table-subscript.R

```diff
@@ -312,6 +313,45 @@ updateIsSubscriptedAttr <- function(y, x) {
     y
 }
 
+updateCellText <- function(y, x.attributes, evaluated.args) {
+    cell.text <- x.attributes$celltext
+    if (!is.array(cell.text)) {
+        return (y)
+    }
+
+    # convert row/column/stat name args to numeric indices
+    indices <- lapply(seq_along(evaluated.args), function(i) {
+        if (is.character(evaluated.args[[i]]) && !is.null(x.attributes$dimnames)) {
+            dim.names <- x.attributes$dimnames[[i]]
+            all.indices <- seq_along(dim.names)
+            names(all.indices) <- dim.names
+            all.indices[evaluated.args[[i]]]
+        } else {
+            evaluated.args[[i]]
+        }
+    })
```

Comment:
Is this necessary? Are the dimnames of `celltext` not aligned to the input table?

Reply:
I don't set the dimnames of celltext (it is tricky to do so as the data sometimes becomes out of sync when setting the attribute). It is a lot easier to do this.